objects/src/accounts/storage/mod.rs

```diff
@@ -266,6 +271,10 @@ impl AccountStorage {
             self.set_item(slot_idx, slot_value)?;
         }
 
+        for &(slot_idx, ref map_delta) in delta.updated_maps.iter() {
+            self.set_map_item(slot_idx, map_delta.clone())?;
+        }
```

Comment:
Instead of doing it this way, I'd probably add `StorageMap::apply_delta()` method to handle what is now handled in `set_map_item()`.